src/main/java/qna/domain/Question.java

```diff
@@ -47,12 +58,27 @@ public Question(Long id, String title, String contents) {
         this.id = id;
         this.title = title;
         this.contents = contents;
+        this.answers = new Answers();
     }
 
     public Question writeBy(User writer) {
         this.writer = writer;
         return this;
     }
+    public List<DeleteHistory> deleteByOwner(User owner) throws CannotDeleteException {
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
+        deleteHistories.add(deleteQuestion(owner));
+        deleteHistories.addAll(answers.deleteAllByOwner(owner));
+        return deleteHistories;
+    }
+
+    private DeleteHistory deleteQuestion(User user) throws CannotDeleteException {
+        if (!isOwner(user)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+        setDeleted(true);
+        return new DeleteHistory(ContentType.QUESTION, this.getId(), this.getWriter(), LocalDateTime.now());
```

Comment:
위에서도 의견 드렸지만 이 부분도 `factory method`가 있으면 더 좋겠습니다 😄 

```suggestion
        return DeleteHistory.question(this.getId(), this.getWriter());
```